go/master/client.go

```diff
@@ -24,15 +18,25 @@ type Client struct {
 //
 // bufSize is the record buffer size. NextRecord will read from this
 // buffer.
-func NewClient(addr Addresser, bufSize int) *Client {
+func NewClient(addrCh <-chan string, bufSize int) *Client {
 	c := &Client{}
 	c.conn = connection.New()
 	c.ch = make(chan []byte, bufSize)
-	go c.monitorMaster(addr)
+	go c.monitorMaster(addrCh)
 	go c.getRecords()
 	return c
 }
 
+// NewEtcdMasterClient creates a new Client by etcd
+func NewEtcdMasterClient(db DatabaseOperator, bufSize int) *Client {
```

Comment:
`db` is only used here to create `ch`, which is used as input for `NewClient`. this is a sign that maybe we don't need the function `NewEtcdMasterClient`.
I think we need to follow rule: less code is better.
Here is a possible way to do it:
```Go
func paddle_new_master_client(...) {
  p := C.GoString(etcdEndpoints)
  e, err := master.NewEtcdClientWithoutLock(strings.Split(p, ","))
  if err != nil {
    panic(err)
  }
  ch := make(chan string, 1)
  ch <- string(db.WaitMasterReady(DefaultAddrPath, 3))
  go db.WatchWithKey(DefaultAddrPath, ch)
  c := master.NewClient(ch, bufSize)
  return add(c)
}
```